src/math/trig.c

```diff
@@ -210,10 +200,10 @@ int32_t is_scalar_cordic_asin(int32_t sinvalue, int16_t numiters)
 	 */
 	if ((sinvalue >> 1) > cord_arcsincos_q28fl) {
 		x = 0;
-		y = cord_arcsincos_q30fl;
+		y = CORD_ARCSINCOS_Q2_30;
 		z = PI_DIV2_Q3_29;
 	} else {
-		x = cord_arcsincos_q30fl;
+		x = CORD_ARCSINCOS_Q2_30;
 		y = 0;
 		z = 0;
 	}
```

Comment:
just noticed some code below:
```
	if (j >= 31)
		j = 31;
```
hopefully the compiler catches this, but you obviously don't need to assign 31, if `j` is already 31, so the condition should be `>`, not `>=` Also, if it's important to optimise every bit of it, you could use the fact, that if `b_i > 31` then `j` is definitely `> 31` too. You could also use `__buiktin_expect()` (often wrapped in `likely()` / `unlikely()` style macros) to help the compiler optimise for more likely branches.